client/src/main/scala/org/http4s/armeria/client/ArmeriaClient.scala

```diff
@@ -103,17 +103,21 @@ private[armeria] final class ArmeriaClient[F[_]] private[client] (
     } yield Response(status = status, headers = toHeaders(headers), body = body)
   }
 
-  /** Converts [[java.util.concurrent.CompletableFuture]] to `F[_]` */
-  private def fromCompletableFuture(
-      completableFuture: CompletableFuture[ResponseHeaders]): F[ResponseHeaders] =
-    F.async_[ResponseHeaders] { cb =>
-      val _ = completableFuture.handle { (result, ex) =>
-        if (ex != null)
-          cb(Left(ex))
-        else
-          cb(Right(result))
-        null
+  private def fromCompletableFuture(cf: => CompletableFuture[ResponseHeaders]): F[ResponseHeaders] =
+    F.async { cb =>
+      cf.handle[Unit] { (result: ResponseHeaders, err: Throwable) =>
+        err match {
+          case null =>
+            cb(Right(result))
+          case _: CancellationException =>
+            ()
```

Comment:
This special case doesn't exist in Cats Effect.  Does the test fail again without it?